src/sortStudents.ts

```diff
@@ -1,16 +1,116 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
+}
+
+export type SortOrder = 'asc' | 'desc';
+
+function averageGrade(array: number[]): number {
+  const result: number = array.reduce((accum, current) => accum + current,
+    0);
+
+  return result / array.length;
 }
 
-// create SortOrder type
-export type SortOrder;
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): object[] {
+  let preparedArray: object[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        preparedArray = preparedArray.sort(
+          (student1, student2) => student1.name.localeCompare(student2.name),
+        );
+      }
+
+      if (order === 'desc') {
+        preparedArray = preparedArray.sort(
+          (student1, student2) => student2.name.localeCompare(student1.name),
+        );
+      }
+
+      break;
+
+    case SortType.Surname:
+      if (order === 'asc') {
+        preparedArray = preparedArray.sort(
+          (student1, student2) => student1.surname
+            .localeCompare(student2.surname),
+        );
+      }
+
+      if (order === 'desc') {
+        preparedArray = preparedArray.sort(
+          (student1, student2) => student2.surname
+            .localeCompare(student1.surname),
+        );
+      }
+
+      break;
```

Comment:
try to merge same cases to reduce the code
```suggestion
    case SortType.Name:
    case SortType.Surname:
      if (order === 'asc') {
        preparedArray = preparedArray.sort(
          (student1, student2) => student1[sortBy].localeCompare(student2[sortBy]),
        );
      }

      if (order === 'desc') {
        preparedArray = preparedArray.sort(
          (student1, student2) => student2[sortBy].localeCompare(student1[sortBy]),
        );
      }

      break;
```